crates/ruff_python_parser/src/semantic_errors.rs

```diff
@@ -72,6 +72,22 @@ impl SemanticSyntaxChecker {
                     Self::duplicate_type_parameter_name(type_params, ctx);
                 }
             }
+            Stmt::Assign(ast::StmtAssign { targets, .. }) => {
+                // test_err single_starred_assignment_target
+                // *a = (1,)
+
+                // test_ok single_starred_assignment_target
+                // (*a,) = (1,)
+                // *a, = (1,)
+                // [*a] = (1,)
+                if let [Expr::Starred(ast::ExprStarred { range, .. })] = targets.as_slice() {
+                    Self::add_error(
```

Comment:
I think I'd find it useful to keep the `test_err` closer to the `add_error` calls, so I'd switch the `test_err` and `test_ok` comments around and move them right above the `Self::add_error` call:
```rs
if let ... {
    // test_ok ...

    // test_err ...
	Self::add_error(...)
}
```